src/context/CredentialsContext.js

```diff
@@ -37,20 +37,13 @@ export const CredentialsProvider = ({ children }) => {
 		setVcEntityList(vcEntityList);
 	};
 
-	const pollForCredentials = () => {
+	const pollForCredentials = useCallback(() => {
 		let attempts = 0;
 		isPolling.current = true;
 
 		intervalId.current = setInterval(async () => {
```

Comment:
Since this function unconditionally sets `isPolling` and `intervalId` every time it's called, and those refs aren't used anywhere outside this function, I don't think you need the refs for this. Would you like me to push a commit that refactors the function to just use local variables instead? (I haven't tested to verify that it works)

Reply:
Ok, here it is: https://github.com/wwWallet/wallet-frontend/commit/7935a2e245a42521bbbacdfb1c72f8197dc0bc56

I haven't tested it and I'm not sure how to, since I'm not very familiar with this code. Do you mind if I leave testing to you?

Reply:
Works correct :smiley:

Reply:
feel free to push it

Reply:
You can do it too on your end :slightly_smiling_face: `git merge --no-ff origin/credentialscontext-no-ref`

Reply:
Oh right, I forgot to mention the branch name, I only gave you the commit :facepalm:

Reply:
So, ready for merge!